speedtest/test_performance_openssl.py

```diff
@@ -0,0 +1,247 @@
+        vcdir = os.path.join("c:\\", "Program Files (x86)",
+                             "Microsoft Visual Studio {}".format(vc),
+                             "VC", "bin")
+        vcvars = os.path.join(vcdir, "vcvars32.bat")
+        if os.path.exists(vcvars):
+            return vcdir, vcvars
+
+    raise Exception("cannot find visual studio!")
+
+
+def configure_openssl():
+    """
+    Run the configure steps (requires perl)
+    :return:
+    """
+    with BlockMessage("configure openssl"):
+        subprocess.check_call(["perl",
+                               "Configure", "VC-WIN32", "no-asm", "--prefix=c:\openssl"],
+                              env=ENVS,
+                              cwd=SOURCES[0])
+
+    with BlockMessage("generate makefiles"):
+        subprocess.check_call([os.path.join("ms", "do_ms.bat")],
+                              shell=True,
+                              env=ENVS,
+                              cwd=SOURCES[0])
+
+
+def setup_function(request):
+    """
+    Ensure a clean build tree before each test
+    :return:
+    """
+    clean_openssl_build()
+    configure_openssl()
+
+
+def setup_module():
+    """
+    Check that our exe has been built.
+    :return:
+    """
+    if not os.path.isfile(CLCACHE):
+        pytest.fail("please build the exe first")
+
+    _, vcvars = find_visual_studio()
+    with BlockMessage("getting vc envs"):
+        getenvs = subprocess.check_output([os.path.join(THISDIR, "get_vc_envs.bat"), vcvars])
+        for line in getenvs.splitlines():
+            if "=" in line:
+                name, val = line.split("=", 1)
+                ENVS[name.upper()] = val
+
+    download_openssl()
+
+
+def replace_wipe_cflags(filename):
+    """
+    Open the nmake file given and turn off PDB generation for .obj files
+    :param filename:
+    :return:
+    """
+    lines = []
+    with open(filename, "rb") as makefile:
+        for line in makefile.readlines():
+            if line.startswith("APP_CFLAG="):
+                lines.append("APP_CFLAG=")
+            elif line.startswith("LIB_CFLAG="):
+                lines.append("LIB_CFLAG=/Zl")
+            else:
+                lines.append(line.rstrip())
+
+    with open(filename, "wb") as makefile:
+        for line in lines:
+            makefile.write(line + "\r\n")
+
+
+def build_openssl(addpath=None, envs=ENVS, pdbs=False):
+    """
+    Build openssl, optionally prefixing addpath to $PATH
+    :param addpath:
+    :param envs: env var dict to use
+    :param pdbs: if False, turn off pdb generation in the makefile
+    :return:
+    """
+    nmakefile = os.path.join("ms", "nt.mak")
+    if not pdbs:
+        replace_wipe_cflags(os.path.join(SOURCES[0], nmakefile))
+
+    if addpath is not None:
+        envs["PATH"] = addpath + os.pathsep + envs["PATH"]
+
+    try:
+        with BlockMessage("running nmake"):
+            subprocess.check_output(["nmake", "-f", nmakefile],
+                                    shell=True,
```

Comment:
Actually it doesn't here, but I found that check_call() doesn't process any supplied env vars unless shell=True, this used to be a check_call()